compendium/DeclarativeServices/test/gtest/TestComponentConfigurationImpl.cpp

```diff
@@ -981,6 +983,96 @@ namespace cppmicroservices
+            auto logger = std::make_shared<SCRLogger>(frameworkContext);
+            auto extRegistry = std::make_shared<SCRExtensionRegistry>(logger);
+
+            auto asyncWorkService = std::make_shared<SCRAsyncWorkService>(frameworkContext, logger);
+            auto notifier
+                = std::make_shared<ConfigurationNotifier>(frameworkContext, fakeLogger, asyncWorkService, extRegistry);
+
+            // Publish ConfigurationListener
+            auto configListener
+                = std::make_shared<cppmicroservices::service::cm::ConfigurationListenerImpl>(frameworkContext,
+                                                                                             logger,
+                                                                                             notifier);
+            auto configListenerReg
+                = frameworkContext.RegisterService<cppmicroservices::service::cm::ConfigurationListener>(
+                    std::move(configListener));
+
+#    if defined(US_BUILD_SHARED_LIBS)
+            auto cabundles = frameworkContext.InstallBundles(test::GetConfigAdminRuntimePluginFilePath());
+            for (auto& bundle : cabundles)
+            {
+                bundle.Start();
+            }
+#    endif
+
+            // Get a service reference to ConfigAdmin to create the configuration object.
+            auto configAdminServiceRef
+                = frameworkContext.GetServiceReference<cppmicroservices::service::cm::ConfigurationAdmin>();
+            ASSERT_TRUE(configAdminServiceRef) << "GetService failed for ConfigurationAdmin.";
+            auto configAdminService = frameworkContext.GetService(configAdminServiceRef);
+            ASSERT_TRUE(configAdminService);
+
+            compMetadata->serviceMetadata.interfaces = { us_service_interface_iid<dummy::ServiceImpl>() };
+            compMetadata->immediate = true;
+            compMetadata->configurationPolicy = "require";
+            compMetadata->configurationPids = { "sample::config" };
+
+            auto compConfig = std::make_shared<SingletonComponentConfigurationImpl>(compMetadata,
+                                                                                    framework,
+                                                                                    mockRegistry,
+                                                                                    fakeLogger,
+                                                                                    notifier);
+            auto mockCompContext = std::make_shared<MockComponentContextImpl>(compConfig);
+            auto mockCompInstance = std::make_shared<MockComponentInstance>();
+            compConfig->SetComponentInstancePair(InstanceContextPair(mockCompInstance, mockCompContext));
+
+            EXPECT_CALL(*mockCompInstance, Modified()).Times(0);
+            EXPECT_CALL(*mockCompInstance, DoesModifiedMethodExist()).WillRepeatedly(testing::Return(true));
```

Comment:
There should be another test/testpoint that checks a component instances without a `Modified` method implementation also does the right thing (i.e. the service component is not destroyed and created again).

Reply:
[Fixed](https://github.com/CppMicroServices/CppMicroServices/pull/985/commits/e5e53bbd1270621344d13763bc0439f69122c135 ). No new test needed